### PR TITLE
updates security considerations

### PR DESCRIPTION
**Pull request by kriskwiatkowski**

### CODE REVIEW

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 1]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
-The security of the described key exchange relies on the security, in particular
-the collision resistance, of the used key-derivation function.
+Security of SIDH is based on the isogeny walk problem, assuming elliptic
+curves between isogenies are supersingular (see [SIKE] chapter 4.1).
+Algorithms solving this problem as well as usage of isogenies as drop-in
+replacement for Diffie-Hellman are relatively young area of research.
+Therefore the security of the ECDHE-SIDH handshake does not rely on the
+security of SIDH exclusively.
+
+Idea behind ECDHE-SIDH hybrid scheme is to combine existing key-agreement
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: "behind the ECDHE-SIDH"

Comment (franziskuskiefer) [reply draft-kiefer-tls-ecdhe-sidh.md]: "to combine an existing"

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 2]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+algorithm with what's belived to be quantum-resistant one. Large quantum
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: "believed to be a"

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 3]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+computers will be able to break both x25519 and x448. In these case ECDHE=SIDH
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: Maybe something like "When large quantum computers are available" ...?

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 4]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+scheme is still safe assuming security of SIDH. From the other hand if SIDH is
+found to be flawed, scheme is still secure on classical computer assuming
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: "the hybrid scheme" ... "against classical attacks" ...

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 5]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+security of x25519/x448. Security estimates for classical and quantum
+computers are provided in table below based on {{SIKE}} and {{RFC7748}}.
+
+| Scheme        | Classical | Quantum | NIST PQ category |
+|---------------|-----------|---------|------------------|
+| x25519sidh503 | 128-bit   | 128-bit |      1           |
+| x448sidh751   | 224-bit   | 192-bit |      3           |
+
+As described in [ISOSEC] it is possible to perform active attack for
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: "active attacks on"

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 6]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+static-static or non-interactive variants of SIDH scheme. The countermeasure
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: "of the SIDH scheme."

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 7]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+for this attack was already described in [KLM15]. Research proposes so-called
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: ~already~

File: draft-kiefer-tls-ecdhe-sidh.md [hunk 8]
```diff
@@ -209,14 +238,41 @@ specified in Section 9 of {{eSIDH}}.
...
+"indirect key validation", using Fujisaki-Okamoto type transform.
+Nevertheless, using this transform is impractical and thus SIDH can be
+considered secure only if used for ephemeral exchanges. More detailed
```
Comment (franziskuskiefer) [inline draft-kiefer-tls-ecdhe-sidh.md]: Maybe "used with ephemeral values."?

Comment (franziskuskiefer) [reply draft-kiefer-tls-ecdhe-sidh.md]: "A more detailed" ...

**Review by franziskuskiefer — COMMENTED** [review]
Thanks. Looks good. I added a couple comments. Mostly editorial/language.

File: draft-kiefer-tls-ecdhe-sidh.md (hunk 8 above)
Comment (kriskwiatkowski) [reply draft-kiefer-tls-ecdhe-sidh.md]: Let's make it "ephemeral keys", will be clear

**Comment by kriskwiatkowski** [comment]
Great, thanks for your comments Franziskus.

**Pull request merged**